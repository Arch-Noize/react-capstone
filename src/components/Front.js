import React from 'react';
import { useSelector } from 'react-redux';
import wotc from '../assets/wotc-srd.jpg';
import tob from '../assets/tob.jpg';
import tob2 from '../assets/tob2.jpg';
import tob3 from '../assets/tob3.jpg';
import menagerie from '../assets/menagerie.jpg';
import cc from '../assets/cc.jpg';
import styles from '../styles/Front.module.css';

const Front = () => {
  const { source, monsters } = useSelector((state) => state.monster);

  const sourceImgs = {
    'wotc-srd': wotc,
    tob,
    tob2,
    tob3,
    menagerie,
    cc,
  };

  return (
    <div className={styles.front}>
      <img src={sourceImgs[source]} alt={source} />
      <div>
        <h1>{source.toUpperCase()}</h1>
        <p>
          {monsters[0].count}
          {' '}
          Creatures
        </p>
      </div>
    </div>
  );
};

export default Front;
