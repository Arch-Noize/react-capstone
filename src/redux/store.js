import { configureStore } from '@reduxjs/toolkit';
import monsterReducer from './class/monsterSlice';

export default configureStore({
  reducer: {
    monster: monsterReducer,
  },
});
