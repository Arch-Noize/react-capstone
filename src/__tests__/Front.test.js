import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Front from '../components/Front';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('renders Front book component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Front />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
