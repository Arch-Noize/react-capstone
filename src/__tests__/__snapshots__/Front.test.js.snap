// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Front book component matches the snapshot 1`] = `
<div>
  <div
    class="front"
  >
    <img
      alt=""
      src="all.jpg"
    />
    <div>
      <h1>
        All Books
      </h1>
      <p>
        0
        +
         
        Creatures
      </p>
    </div>
  </div>
</div>
`;
